### PR TITLE
Add version and help to the cli options

### PR DESCRIPTION
**Pull request by robinraju**
Add `--version` and `--help` to CMT applications.

It can now perform the following:

```
$ cmt-delinearize --version
Course Management Tools: delinearize 1.0.0

$ cmt-linearize --version
Course Management Tools: linearize 1.0.0

$ cmt-mainadm --version
Course Management Tools: mainadm 1.0.0

$ cmt-studentify --version
Course Management Tools: studentify 1.0.0
```

The `--help` option prints the usage info

```
$ cmt-mainadm --help
Course Management Tools: mainadm 1.0.0
Usage: mainadm [options] mainRepo

  mainRepo                 base folder holding main course repository
  -mjvm, --multi-jvm       generate multi-jvm build file
  -b, --build-file-regen   regenerate project root build file
  -d, --delete <value>
  -r, --renumber           renumber exercises
  -ro, --renumber-offset <value>
                           renumber exercises - offset
  -rs, --renumber-step <value>
                           renumber exercises - step
  -dib, --duplicate-insert-before <value>

  -cfg, --config-file <value>
                           configuration file
  -c, --check-main-repo    verify soundness of main repository
  -amc, --add-main-commands
                           add command files to main repository
  -t, --generate-tests-script <value>
                           generate a script that tests main repo, studentified repo functionality and linearize/delinearize
  -g, --init-studentified-test-repo-as-git
                           Generate studentified repo as a git repo in tests
  -dot, --dotty            studentified repository is a Dotty project
  -nar, --no-auto-reload-sbt
                           no automatic reload on build definition change
  --help                   Prints the usage text
  -v, --version            Prints the version info
```

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -7,11 +7,17 @@ lazy val `course-management-tools` =
       delinearize,
       mainadm
     )
-    .settings(ThisBuild / scalaVersion := Version.scalaVersion)
+    .settings(CommonSettings.commonSettings: _*)
+    .settings(skip in publish := true)
```
Comment (eloots) [inline build.sbt:11]: I have noticed that removing `.settings(ThisBuild / scalaVersion := Version.scalaVersion)` in another project (a Dotty project, to be precise), caused `scalaVersion` not to be picked up in sub-projects. You can see that when starting a REPL session in a sub-project.

File: linearize/src/main/scala/com/lightbend/coursegentools/Linearize.scala [hunk 2]
```diff
@@ -33,7 +33,14 @@ object Linearize {
...
     if (cmdOptions.isEmpty) System.exit(-1)
-    val LinearizeCmdOptions(mainRepo, linearizedOutputFolder, multiJVM, forceDeleteExistingDestinationFolder, configurationFile, isADottyProject, autoReloadOnBuildDefChange, bareLinRepo) = cmdOptions.get
+    val LinearizeCmdOptions(mainRepo,
+                           linearizedOutputFolder,
+                           multiJVM,
+                           forceDeleteExistingDestinationFolder,
+                           configurationFile,
+                           isADottyProject,
+                           autoReloadOnBuildDefChange,
+                           bareLinRepo) = cmdOptions.get
```
Comment (eloots) [inline linearize/src/main/scala/com/lightbend/coursegentools/Linearize.scala:43]:
```suggestion
                            linearizedOutputFolder,
                            multiJVM,
                            forceDeleteExistingDestinationFolder,
                            configurationFile,
                            isADottyProject,
                            autoReloadOnBuildDefChange,
                            bareLinRepo) = cmdOptions.get
```

Comment (eloots) [reply linearize/src/main/scala/com/lightbend/coursegentools/Linearize.scala:43]: Minor alignment issue

Comment (eloots) [reply linearize/src/main/scala/com/lightbend/coursegentools/Linearize.scala]: Resolved by [c78806c](https://github.com/eloots/course-management-tools/pull/121/commits/c78806c50382327278563a021fbf2459561ad5e5)

File: build.sbt (hunk 1 above)
Comment (robinraju) [reply build.sbt:11]: `scalaVersion` is applied to all sub projects as follows:
```
sbt:course-management-tools> scalaVersion
[info] studentify / scalaVersion
[info] 	2.13.3
[info] mainadm / scalaVersion
[info] 	2.13.3
[info] delinearize / scalaVersion
[info] 	2.13.3
[info] linearize / scalaVersion
[info] 	2.13.3
[info] core / scalaVersion
[info] 	2.13.3
[info] scalaVersion
[info] 	2.13.3
```
I can confirm this doesn't affect the course templates and is specific to some dotty projects

**Pull request merged**